lib/dill/widgets/array_value.rb

```diff
@@ -0,0 +1,18 @@
+module Dill
+  class ArrayValue < Array
+    def downcase_all
+      result = self.map do |item|
+        case item
+        when String
+          item.downcase
+        when Array
+          self.class.new(item).downcase_all
+        when Hash
+          HashValue.new(item).downcase_all
+        end
+      end
+
+      ArrayValue.new(result)
+    end
+  end
+end
```

Comment:
Do you think this class is necessary? Would making `downcase_all` a private method in the `CucumberMethods` module be enough?